tests/ui/nop_match.rs

```diff
@@ -0,0 +1,103 @@
+// run-rustfix
+#![warn(clippy::nop_match)]
+#![allow(clippy::manual_map)]
+#![allow(dead_code)]
+
+#[derive(Clone, Copy)]
+enum Choice {
+    A,
+    B,
+    C,
+    D,
+}
+
+fn useless_match(x: i32) {
```

Comment:
How about adding a test case with a `x => x` arm?

Reply:
That's a good point, I didn't check for that in my code. :D 

Also, that reminds me, should I be checking `ref x => *x` as well? Or is that too much?